lib/tasks/bundlers/generateComponentPreload.js

```diff
@@ -51,37 +60,63 @@ module.exports = function({workspace, dependencies, taskUtil, options: {projectN
 					`found for project: ${projectName}`);
 			}
 
-			return Promise.all(allNamespaces.map((namespace) => {
+			const allFilterExcludes = negateFilters(excludes);
+			const unusedFilterExcludes = new Set(allFilterExcludes);
+
+			const bundleDefinitions = allNamespaces.map((namespace) => {
 				const filters = [
 					`${namespace}/`,
 					`!${namespace}/test/`,
 					`!${namespace}/*.html`
 				];
 
-				// Exclude other namespaces
+				// Add configured excludes for namespace
+				allFilterExcludes.forEach((filterExclude) => {
+					// Allow all excludes and limit re-includes to the component namespace
+					if (filterExclude.startsWith("!") || filterExclude.startsWith(`${namespace}/`)) {
+						filters.push(filterExclude);
+						unusedFilterExcludes.delete(filterExclude);
+					}
+				});
+
+				// Exclude other namespaces at the end of filter list to override potential re-includes
+				// from "excludes" config
 				allNamespaces.forEach((ns) => {
```

Comment:
I think, I did not fully understand what this second step is needed for.

Reply:
For example a root-component's preload should not contain any resources of the configured sub-components.

This exclude must happen after the configured preload-excludes have been applied, so that the latter can't re-include any resources from the sub-components. While that could have valid use cases, we think it would be unwanted in most cases.